tools/flang2/flang2exe/lldebug.cpp

```diff
@@ -2764,6 +2770,66 @@ lldbg_create_assumed_len_string_type_mdnode(LL_DebugInfo *db, SPTR sptr,
   return llmd_finish(mdb);
 }
 
+INLINE static LL_MDRef
+lldbg_create_deferred_len_string_type_mdnode(LL_DebugInfo *db, SPTR sptr,
+                                            int findex)
+{
+  LL_MDRef mdLen = ll_get_md_null();
+  LL_MDRef mdLenExp = ll_get_md_null();
+  LLMD_Builder mdb = llmd_init(db->module);
+  const char *name = "character(*)";
+  const long long size = 32;
+  const long long alignment = 0;
+  const int encoding = 0;
+
+  if (SDSCG(REVMIDLNKG(sptr)) && (DTY(DTYPEG(sptr)) == TY_PTR)) {
```

Comment:
If the SDSC is not available, what kind of debug info will the fallback path below generate for such a deferred length string? Can the user print the string inside a debugger?

Reply:
@bryanpkc, thanks for the review comment, i will resolve the NIT's. 
assumed length and deferred length string representation is different in flang, deferred length string is represented with $sd and $p (like array), hence SDSC(descriptors) will always be associated with the deferred length string variables. As you mentioned, in any case if string length information could not be fetched in lldbg_create_deferred_len_string_type_mdnode(), then DIStringType wont emit the stringLength field itself. and this is the fallback path as you mentioned.

Reply:
Thanks for the clarification.